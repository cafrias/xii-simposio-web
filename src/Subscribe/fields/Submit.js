// @flow
import React from 'react'

const SubmitButton = () =>
  <div className="control">
    <button className="button is-link" type="submit">Submit</button>
  </div>

export default SubmitButton
