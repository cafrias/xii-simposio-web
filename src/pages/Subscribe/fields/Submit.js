// @flow
import React from 'react'

type SubmitButtonProps = {
  id: string,
  loading: boolean
}

const SubmitButton = ({id, loading}: SubmitButtonProps) => {
  const className = `
    button is-link
    ${loading ? 'is-loading' : ''}
    `

  return (
    <div className="control">
      <button id={id} className={className} type="submit">
        Enviar
      </button>
    </div>
  )
}

export default SubmitButton
