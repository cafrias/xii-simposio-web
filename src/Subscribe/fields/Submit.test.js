import React from 'react'
import { shallow } from 'enzyme'

import Submit from './Submit'

const reqProps = {
  id: 'id',
  loading: false,
  success: false,
  failure: false,
}

test('renders with basic props', () => {
  const wrapper = shallow(<Submit {...reqProps} />)

  expect(wrapper.find(`#${reqProps.id}`)).toHaveLength(1)
})

test('matches snapshot', () => {
  const wrapper = shallow(<Submit {...reqProps}/>)

  expect(wrapper).toMatchSnapshot()
})

test('renders loading class', () => {
  const props = Object.assign({}, reqProps, {
    loading: true
  })
  const wrapper = shallow(<Submit {...props}/>)

  expect(wrapper.find('.is-loading')).toHaveLength(1)
})

test('renders error theme', () => {
  const props = Object.assign({}, reqProps, {
    failure: true
  })
  const wrapper = shallow(<Submit {...props}/>)

  expect(wrapper.find('.is-danger')).toHaveLength(1)
  expect(wrapper.find('.fa-times')).toHaveLength(1)
})

test('renders success theme', () => {
  const props = Object.assign({}, reqProps, {
    success: true
  })
  const wrapper = shallow(<Submit {...props}/>)

  expect(wrapper.find('.is-success')).toHaveLength(1)
  expect(wrapper.find('.fa-check')).toHaveLength(1)
})
