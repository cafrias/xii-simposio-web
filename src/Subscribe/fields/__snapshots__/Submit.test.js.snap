// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="control"
>
  <button
    className="button is-link "
    id="id"
    type="submit"
  >
    Submit
  </button>
</div>
`;
